client/landing/stepper/declarative-flow/internals/analytics/record-design.ts

```diff
@@ -101,9 +109,17 @@ export function getVirtualDesignProps( design: Design, styleVariation?: StyleVar
 		variationSlugSuffix = `-${ design.preselected_style_variation.slug }`;
 	}
 
+	/**
+	 * If the design is virtual, and it has a recipe with pattern_ids,
+	 * then we assume that it's a pattern based virtual theme.
+	 */
+	const virtual_theme_pattern =
+		design.is_virtual && design.recipe?.pattern_ids ? design.recipe?.pattern_ids[ 0 ] : null;
```

Comment:
It might be safer to also check the length of pattern_ids:
```suggestion
		design.is_virtual && design.recipe?.pattern_ids?.length ? design.recipe?.pattern_ids[ 0 ] : null;
```

Reply:
Done! :)